mapbox/dr-ui
# [info arch] Make NavigationAccordion toggleable

This PR makes a proposal on how to handle the NavigationAccordion items on smaller devices ref https://github.com/mapbox/dr-ui/pull/344#issuecomment-721859813 and if accepted will merge into #344. This PR makes NavigationAccordion sub page menu's toggleable and will collapse the items on smaller devices be default:

- Separates the title/link and the chevron icon. The chevron icon is now a button that controls the sub page menu. The user can toggle a section's sub page menu from any page. 
- A section's sub page menu opens automatically when the section or any of its sub pages are active. However, on smaller devices all toggles are collapsed on load, but the user can toggle to reveal the subpages.
- On smaller devices, the padding for the NavigationAccordion links increases slightly to improve tap target sizes.

## Screenshots

> ![image](https://user-images.githubusercontent.com/2180540/98396971-9b3c4400-202c-11eb-9d69-770c1f0b0da5.png)
> Sub page menu toggle collapses on mobile by default

> ![2020-11-06 at 12 34 PM](https://user-images.githubusercontent.com/2180540/98397070-c888f200-202c-11eb-9715-806caf4188e4.gif)
> Toggle a section's sub page menu from anywhere


## How to test

- Visit the catalog site (`npm run start-docs` -> http://localhost:8080/dr-ui/guides/)
    - From any page, toggle the "Guides" chevron icon. It should expand and collapse its sub page menu.
    - Visit http://localhost:8080/dr-ui/guides/ on desktop, the sub page menu for "Guides" should open on load.
    - Visit http://localhost:8080/dr-ui/guides/ on mobile, the sub page menu for "Guides" should not open on load, but can be open if you press the chevron icon.

## QA checklist




Open the test cases app locally on:
